aten/src/ATen/native/ReflectionPadding1d.cpp

```diff
@@ -0,0 +1,227 @@
+    scalar_t * grad_input, scalar_t * grad_output,
+    int64_t nplane,
+    int64_t input_w, int64_t output_w,
+    int64_t pad_l) {
+  int64_t i_start_x = std::max(int64_t(0), -pad_l);
+  int64_t o_start_x = std::max(int64_t(0), pad_l);
+
+  int64_t k, ip_x;
+#pragma omp parallel for private(k, ip_x)
+
+  for (k = 0; k < nplane; k++) {
+    for (int64_t j = 0; j < output_w; j++) {
+      if (j < pad_l) {
+        ip_x = pad_l * 2 - j;
+      } else if (j >= pad_l && j < input_w + pad_l) {
+        ip_x = j;
+      } else {
+        ip_x = (input_w + pad_l - 1) * 2 - j;
+      }
+      ip_x = ip_x - o_start_x + i_start_x;
+
+      scalar_t *src_p = grad_output + k*output_w + j;
+      scalar_t *dest_p = grad_input + k*input_w + ip_x;
+      *dest_p += *src_p;
+    }
+  }
+}
+
+void reflection_pad1d_backward_out_template(
+    Tensor& grad_input, Tensor const& grad_output_, Tensor const& input,
+    IntList padding) {
+  int64_t dim_plane = 0;
+  int64_t dim_w = 1;
+  int64_t nbatch = 1;
+
+  if (input.ndimension() == 3) {
+    nbatch = input.size(0);
+    dim_w++;
+    dim_plane++;
+  }
+
+  /* sizes */
+  auto pad_l = padding[0];
+  auto pad_r = padding[1];
+  int64_t nplane = input.size(dim_plane);
+  int64_t input_w = input.size(dim_w);
+  int64_t output_w  = input_w + pad_l + pad_r;
+
+  AT_CHECK(output_w == grad_output_.size(dim_w), "grad_output width unexpected."
+    " Expected: ", output_w, ", Got: ", grad_output_.size(dim_w));
+
+  /* get contiguous gradOutput */
+  Tensor grad_output = grad_output_.contiguous();
+
+  /* backprop */
+  if (input.ndimension() == 2) {
+    AT_DISPATCH_FLOATING_TYPES_AND_HALF(
+      input.type(), "reflection_pad1d_backward", [&] {
+        reflection_pad1d_backward_out_frame(
+          grad_input.data<scalar_t>(), grad_output.data<scalar_t>(),
+          nplane,
+          input_w, output_w,
+          pad_l);
+        }
+    );
+  } else {
+    int64_t p;
+#pragma omp parallel for private(p)
+    for (p = 0; p < nbatch; p++) {
+      AT_DISPATCH_FLOATING_TYPES_AND_HALF(
+        input.type(), "reflection_pad1d_backward", [&] {
+          reflection_pad1d_backward_out_frame(
+            grad_input.data<scalar_t>() + p * nplane * input_w,
+            grad_output.data<scalar_t>() + p * nplane * output_w,
+            nplane,
+            input_w, output_w,
+            pad_l);
+        }
+      );
+    }
+  }
+}
+} // namespace
+
+Tensor& reflection_pad1d_out_cpu(
+    Tensor& output, const Tensor& input, IntList padding) {
+  reflection_pad1d_out_template(output, input, padding);
+  return output;
+}
+
+Tensor reflection_pad1d_cpu(Tensor const& input, IntList padding) {
+  auto output = at::empty({0}, input.options());
+  reflection_pad1d_out_template(output, input, padding);
+  return output;
+}
+
+Tensor& reflection_pad1d_backward_out_cpu(
+    Tensor& grad_input,
+    const Tensor& grad_output,
+    const Tensor& input,
+    IntList padding) {
+  grad_input = at::zeros_like(input);
```

Comment:
Actually, this doesn't look right -- you aren't overwriting the data in grad_input, you are creating a new tensor and writing to it.  So if you say, create a buffer of the correct size and expect to write the output of this function to it, that won't actually happen.  I think you want resize followed by zero_ instead.